init-tools.sh

```diff
@@ -106,28 +108,32 @@ if [ ! -e "$__DOTNET_PATH" ]; then
                 ;;
 
             *)
-                echo "Unsupported OS '$OSName' detected. Downloading linux-$__PKG_ARCH tools."
+            echo "Unsupported OS '$OSName' detected. Downloading linux-$__PKG_ARCH tools."
                 OS=Linux
                 __PKG_RID=linux
                 ;;
-      esac
+        esac
 
-      __DOTNET_PKG=dotnet-sdk-${__DOTNET_TOOLS_VERSION}-$__PKG_RID-$__PKG_ARCH
+        __DOTNET_PKG=dotnet-sdk-${__DOTNET_TOOLS_VERSION}-$__PKG_RID-$__PKG_ARCH
     fi
-
     mkdir -p "$__DOTNET_PATH"
 
     echo "Installing dotnet cli..."
     __DOTNET_LOCATION="https://dotnetcli.azureedge.net/dotnet/Sdk/${__DOTNET_TOOLS_VERSION}/${__DOTNET_PKG}.tar.gz"
 
     install_dotnet_cli() {
-        echo "Installing '${__DOTNET_LOCATION}' to '$__DOTNET_PATH/dotnet.tar'" >> "$__init_tools_log"
-        rm -rf -- "$__DOTNET_PATH/*"
-        # curl has HTTPS CA trust-issues less often than wget, so lets try that first.
-        if command -v curl > /dev/null; then
-            curl --retry 10 -sSL --create-dirs -o "$__DOTNET_PATH/dotnet.tar" "${__DOTNET_LOCATION}"
+        if [[ -z "${DotNetBootstrapCliTarPath-}" ]]; then
```

Comment:
Does this work? I only know of the ones listed here: https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html, would expect `:-`.

Reply:
See http://wiki.bash-hackers.org/syntax/pe#use_a_default_value

Reply:
Yes it works as a test. You have to read the fine print:

```
When not performing substring expansion, using the form described below (e.g., ‘:-’), Bash tests for a parameter that is unset or null. Omitting the colon results in a test only for a parameter that is unset. Put another way, if the colon is included, the operator tests for both parameter’s existence and that its value is not null; if the colon is omitted, the operator tests only for existence.
```

Also I like http://wiki.bash-hackers.org/syntax/pe#use_a_default_value as a reference, little easier to read.

Reply:
For what it is worth while looking this up there are about 100 different ways to do this check so I'm not sure there is a defacto way but I'm happy to be pointed to a clear guide. The best answer I found was https://stackoverflow.com/questions/3601515/how-to-check-if-a-variable-is-set-in-bash and I couldn't figure out why I would use `+` over `-` so I just stuck with what we already have in this script.

Reply:
Oh, wow, I've been missing that for a while. Thanks for the explanation, and yeah, that wiki is nicer.